spatialos-sdk/src/worker/schema.rs

```diff
@@ -0,0 +1,582 @@
+        }
+    }
+
+    pub fn fields_mut(&mut self) -> SchemaObject {
+        SchemaObject {
+            internal: unsafe { Schema_GetComponentDataFields(self.internal) },
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct SchemaCommandRequest {
+    pub component_id: ComponentId,
+    pub internal: *mut Schema_CommandRequest,
+}
+
+impl SchemaCommandRequest {
+    pub fn new(component_id: ComponentId, command_index: FieldId) -> SchemaCommandRequest {
+        SchemaCommandRequest {
+            component_id,
+            internal: unsafe { Schema_CreateCommandRequest(component_id, command_index) },
+        }
+    }
+
+    pub fn component_id(&self) -> ComponentId {
+        unsafe { Schema_GetCommandRequestComponentId(self.internal) }
+    }
+
+    pub fn command_index(&self) -> FieldId {
+        unsafe { Schema_GetCommandRequestCommandIndex(self.internal) }
+    }
+
+    pub fn object(&self) -> SchemaObject {
+        SchemaObject {
+            internal: unsafe { Schema_GetCommandRequestObject(self.internal) },
+        }
+    }
+
+    pub fn object_mut(&mut self) -> SchemaObject {
+        SchemaObject {
+            internal: unsafe { Schema_GetCommandRequestObject(self.internal) },
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct SchemaCommandResponse {
+    pub component_id: ComponentId,
+    pub internal: *mut Schema_CommandResponse,
+}
+
+impl SchemaCommandResponse {
+    pub fn new(component_id: u32, command_index: u32) -> SchemaCommandResponse {
+        SchemaCommandResponse {
+            component_id,
+            internal: unsafe { Schema_CreateCommandResponse(component_id, command_index) },
+        }
+    }
+
+    pub fn component_id(&self) -> ComponentId {
+        unsafe { Schema_GetCommandResponseComponentId(self.internal) }
+    }
+
+    pub fn command_index(&self) -> FieldId {
+        unsafe { Schema_GetCommandResponseCommandIndex(self.internal) }
+    }
+
+    pub fn object(&self) -> SchemaObject {
+        SchemaObject {
+            internal: unsafe { Schema_GetCommandResponseObject(self.internal) },
+        }
+    }
+
+    pub fn object_mut(&mut self) -> SchemaObject {
+        SchemaObject {
+            internal: unsafe { Schema_GetCommandResponseObject(self.internal) },
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct SchemaObject {
+    internal: *mut Schema_Object,
+}
+
+impl SchemaObject {
+    pub fn field<T: SchemaType>(&self, field: FieldId) -> T::RustType {
+        T::from_field(self, field)
+    }
+
+    pub fn deserialize<T: SchemaObjectType>(&self) -> T {
+        T::from_object(self)
+    }
+}
+
+// =================================================================================================
+// Schema Conversion Traits
+// =================================================================================================
+
+pub trait SchemaType: Sized {
+    type RustType: Sized;
+
+    fn from_field(object: &SchemaObject, field: FieldId) -> Self::RustType;
```

Comment:
Nit: `get_field` to be symmetric with `get_field_list`

Reply:
It might actually be good to change the name of `get_field_list` instead 🤔